File: cpp/experiencias.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main() {

    int N, ratos = 0, sapos = 0, coelhos = 0, qtdCobaias, total;
    char tipo;
    double pratos, psapos, pcoelhos;

    cout << "Quantos casos de teste serao digitados? ";
    cin >> N;

    for(int i = 0; i < N; i++){
        cout << "Quantidade de cobaias: ";
        cin >> qtdCobaias;
        cout << "Tipo de cobaia: ";
        cin >> tipo;

        if(tipo == 'R'){
            ratos = ratos + qtdCobaias;
        }
        else if(tipo == 'S'){
            sapos = sapos + qtdCobaias;
        }
        else{
            coelhos = coelhos + qtdCobaias;
        }
    }

    total = ratos + sapos + coelhos;
    pcoelhos = ((double)coelhos / total) * 100.0;
    pratos = ((double)ratos / total) * 100.0;
    psapos = ((double)sapos / total) * 100.0;

    cout << endl << "RELATORIO FINAL:" << endl;
    cout << "Total: " << total << " cobaias" << endl;
    cout << "Total de coelhos: " << coelhos << endl;
    cout << "Total de ratos: " << ratos << endl;
    cout << "Total de sapos: " << sapos << endl;
    cout << fixed << setprecision(2);
    cout << "Percentual de coelhos: " << pcoelhos << endl;
    cout << "Percentual de ratos: " << pratos << endl;
    cout << "Percentual de sapos: " << psapos << endl;

    return 0;
}

File: python/experiencias.py
N: int
totalRatos: int
totalSapos: int
totalCoelhos: int
qtdCobaias: int
totalCobaias: int
tipo: str
pratos: float
psapos: float
pcoelhos: float

N = int(input("Quantos casos de teste serao digitados? "))

totalRatos = 0
totalSapos = 0
totalCoelhos = 0
for i in range(N):
    qtdCobaias = int(input("Quantidade de cobaias: "))
    tipo = str(input("Tipo de cobaia: "))

    if tipo == 'R':
        totalRatos += qtdCobaias
    elif tipo == 'S':
        totalSapos += qtdCobaias
    else:
        totalCoelhos += qtdCobaias

totalCobaias = totalRatos + totalSapos + totalCoelhos
pcoelhos = (float(totalCoelhos) / totalCobaias) * 100.0
pratos = (float(totalRatos) / totalCobaias) * 100.0
psapos = (float(totalSapos) / totalCobaias) * 100.0

print()
print("RELATORIO FINAL:")
print(f"Total: {totalCobaias} cobaias")
print(f"Total de coelhos: {totalCoelhos}")
print(f"Total de ratos: {totalRatos}")
print(f"Total de sapos: {totalSapos}")
print(f"Percentual de coelhos: {pcoelhos:.2f}")
print(f"Percentual de ratos: {pratos:.2f}")
print(f"Percentual de sapos: {psapos:.2f}")
